### DISCUSSION TITLE
New Alien - Electrical Hull Clingers

### DISCUSSION THREAD

**Discussion by Solbusaur** [FakeFishGames/Barotrauma#9488] [Ideas and suggestions]
![image](https://user-images.githubusercontent.com/63509011/176492732-feb21f2a-e91c-453a-95ab-4ef27dffc800.png)

I was going to call them shock piglets, but the name isn't that important yet.
These hairy molluscs are passive, very slow filter-feeding floaters that can stick to a ship's hull and deliver stunning shocks to creatures and humans alike if they get too close, except eachother. They are super allergic to invasion of their personal space, and they want to be left alone.
They won't try to chase submarines or people, and they travel together in large clusters of 8-15. If a ship collides with a swarm of them, they will automatically latch onto the hull and slow it down a tiny bit, sort of like baby molochs tend to already.

Their shocking habits can disrupt divers if they happen to crowd around airlocks and so are easily destroyed by ranged combat, but killing them on sight isn't strictly necessary. My main push for this creature is allowing players to deliberately scoop up swarms of these if they are lucky enough to find them and use them as a natural barrier against small enemies. Crawlers, raptors and tigerthreshers have to get within biting distance of hulls to breach them, and these clingers can be exploited by smart players to repel said creatures. They can't kill them very easily, just stun and lightly burn them like a discharge coil. The drawback is the extra mass the submarine has to push through the water with, the speed of the submarine is slowed down by having too many.

I also would like to see this to add more monsters that aren't strictly threats to a submarine, the world is more believable when there do exist aliens that're beneath humans in the foodchain

**Comment by MaaiG** [comment]
I really like this idea!

Perhaps they can cause electrical interference in the operation of various submarine systems, lighting, communication systems. The shock of the discharge coil may be so appealing to them that they will grow in size for a while, which will drastically affect the speed of the submarine and make it even more difficult for it to function.

Their presence can be both beneficial and harmful to the submarine. There’s got to be a compromise system for why we sometimes want to shoot them and sometimes want them to stay on the hull of the sub.

After all, from time to time, you must clean the hull of every ship :)

**Comment by cshingle** [comment]
I had the same idea. I was thinking of a few fast moving monsters rather than a cloud. The game play point of something like this is to add some monsters that the crew NEEDS to leave the sub to fight. Right now you can just hide in the sub and gun everything down. 

I have even looked into how to implement a modded creature with the effect. This can be implemented via the leach affect on the piezocrystal. I would remove the ElectricShock to make the "correct" way to fight them be leave the sub and just melee them to death.

    <ItemComponent>
      
      <StatusEffect type="OnActive" target="NearbyItems" targettags="junctionbox" range="3000.0" setvalue="true" stackable="false" delay="0.95" extraload="500" />
      <StatusEffect type="OnActive" target="This" IsActive="false" />
      
      <StatusEffect type="Always" target="This" stackable="false" delay="1.0" IsActive="true" />
      <StatusEffect type="Always" target="This" stackable="false" delay="10.0">
        <Use />
      </StatusEffect>
      <StatusEffect type="Always" target="This">
        <ParticleEmitter particle="ElectricShock" distancemin="100" distancemax="120" particlespersecond="5.0" particleamount="5" emitinterval="2.1" anglemin="0" anglemax="360" scalemin="1.0" scalemax="1.2" />
      </StatusEffect>
    </ItemComponent>
